### ISSUE TITLE
[FEATURE] Show source and download button

### ISSUE THREAD

**Issue by maximlt** [pyscript/pyscript#293]
PyScript as a framework could provide two components that could be easily added to the HTML for creators to easily share their app:

* a *show source* button/ui for the app visitors to see the code powering the app
* a *download* button for the app visitors to download the app locally (maybe with an option for the downloaded file(s) to work in offline mode too, if that's even possible)

Viewing the source of a web page is just a couple of clicks away, but I'd argue that lots of potential future users of PyScript aren't necessarily tech savvy, and it's fine, let's make PyScript dead simple for them.

**Comment by verhulstm** [comment]
its works in offline mode.

you just need some file server. like "python3 -m http.server 8080"

**Comment by verhulstm** [comment]
if i understand you correctly, this is a feature request for the example showcase, not really a feature for PyScript itself?

**Comment by maximlt** [comment]
> if i understand you correctly, this is a feature request for the example showcase, not really a feature for PyScript itself?

Nope, I thought that these two components could be a feature of PyScript itself. Which could then be added to the examples:

* *show source* could do the equivalent of "right-click/view shource", or be more opinionated and open a modal with a  repr displaying the dependencies and code in a nice way.
* I really like how PyScript was introduced in disclaiming that you can write an HTML file, share it with your friends, and it just works. I would hope that *download* could help to achieve that, giving the ability to visitors to download a file and share/archive it easily.

(As for the offline mode, I don't want to fit too much into that feature request so I'll stop there 🙃)

**Comment by verhulstm** [comment]
i feel like this should be a code snippet. a recommended pattern. and it should be common in our examples

i am not sure this should be a built in component.

**Comment by marimeireles** [comment]
Hey @maximlt, thanks for the input! :)
We've been having some discussions around that, on how to create a good set of default UI (both stuff like buttons, but also a default theme for the PyScript GUI etc.).
I think the actual implementation is a bit far because we currently have a lot in our hands and this is not one of the top priorities, but it's definitely in our minds to work on it.
A better place for this discussion might be https://community.anaconda.cloud/c/tech-topics/pyscript/
